### PR TITLE
Relay on Internet Time

### PR DESCRIPTION
**Pull request by fileninja**
Hey guys,
just implemented a small service to get the current time via json time service in the internet, returning the current american time. if it cant call this service for whatever reason it will switch to local time and returning a normal Date Object.
Also replaced all the new Date().getTime() Calls with this new Service.

#57 and #211 should be done with that.

### CODE REVIEW

File: client/app/src/time/TimeService.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    .service('timeService', [TimeService]);
+
+  /**
+   * TimeService
+   * @constructor
+   */
+  function TimeService() {
+    function __getJSON(url) {
+      return new Promise(function(resolve, reject) {
+        var xhr = new XMLHttpRequest();
```
Comment (luciorubeens) [inline client/app/src/time/TimeService.js]: any specific reason to not use $http instead of xhr?

File: client/app/src/time/TimeService.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+          }
+        };
+        xhr.send();
+      });
+    };
+    this.getTime = function() {
+      __getJSON('http://date.jsontest.com/').then(function(data) 
+      {
+        return new Date(Date.parse(data["date"]+" "+data["time"]));
+      });
```
Comment (luciorubeens) [inline client/app/src/time/TimeService.js]: missing a catch(?).. if cant call this url returns local date

File: client/app/src/time/TimeService.js (hunk 1 above)
Comment (fileninja) [reply client/app/src/time/TimeService.js]: No, in this case its the same.. its just my Habit. Should i Change?

File: client/app/src/time/TimeService.js (hunk 2 above)
Comment (fileninja) [reply client/app/src/time/TimeService.js]: Should be handled by the “then“ Error function but will Test again.

File: client/app/src/time/TimeService.js (hunk 1 above)
Comment (luciorubeens) [reply client/app/src/time/TimeService.js]: yes, please. it's better to keep the same standard as other services

**Comment by fileninja** [comment]
@luciorubeens changed to $http and checked the local time fallback which worked fine. So actually it returns GMT-4 (New York Time) which you may want to change depending on the default network time. If it doesnt reach the server for whatever reason, it will fallback to clientside time

File: client/app/src/time/TimeService.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+   * @constructor
+   */
+  function TimeService($http) {
+    this.getTime = function() {
+      $http.get('http://date.jsontest.com/', { timeout: 2000 })
+        .then(function(resp) {
+          var date = new Date(Date.parse(resp.data["date"]+" "+resp.data["time"]));
+          //GMT-4 New York Time
+          date.setHours(date.getHours() - 4);
+          return date
```
Comment (fix) [inline client/app/src/time/TimeService.js]: mmm the issue here is that it return a promise and not the real date, also date should be UTC

Comment (fileninja) [reply client/app/src/time/TimeService.js]: What do u mean with promise?
Every Server which is Not in your Hand is a promise. Gmt is actually the same as utc but the URL can also be changed to a utc one

**Comment by luciorubeens** [comment]
Solved #220

**Pull request closed without merging**